components/application-authenticators/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/SAMLSSOAuthenticator.java

```diff
@@ -427,11 +428,11 @@ private void printPostPage(HttpServletResponse response, String url, String post
                 PrintWriter out = response.getWriter();
                 out.println("<html>");
                 out.println("<body>");
-                out.println("<p>You are now redirected to " + url);
+                out.println("<p>You are now redirected to " + Encode.forHtml(url));
                 out.println(" If the redirection fails, please click the post button.</p>");
-                out.println("<form method='post' action='" + url + "'>");
+                out.println("<form method='post' action='" + Encode.forHtmlAttribute(url) + "'>");
                 out.println("<p>");
-                out.println(postPageInputs);
+                out.println(Encode.forHtml(postPageInputs));
```

Comment:
It' wrong to encode this variable here. It is constructed internally from buildPostPageInputs() methods. Only the untrusted variables should be encoded within that method.